src/internal/utils.ts

```diff
@@ -315,22 +315,29 @@ export function getName<U extends AnyParamConstructor<any>>(cl: U) {
   const ctor: any = getRightTarget(cl);
   const options: IModelOptions = Reflect.getMetadata(DecoratorKeys.ModelOptions, ctor) ?? {};
   const baseName: string = ctor.name;
+  const customName = options.options?.customName;
+
+  if (typeof customName === 'function') {
+    return customName(options);
```

Comment:
maybe add an error if the return of this function is an empty string / not the type that is expected (maybe do it with `assertion`)